.github/workflows/ci.yml

```diff
@@ -204,10 +204,14 @@ jobs:
         with:
           toolchain: ${{ env.RUST_VERSION }}
       - uses: Swatinem/rust-cache@v2
-      - name: 'Build Workspace'
+      - name: Install Protoc
```

Comment:
Seems like there a lot of these additions. Can you explain why these were added? Doesn't seem related to this PR but maybe i'm wrong.

Reply:
@JoshuaBatty that is because protoc (protobuf compiler) is needed in order to build the dependencies needed for the `new-key` and `parse-secret` which were ported from https://github.com/FuelLabs/fuel-core/blob/master/bin/keygen/src/keygen.rs

Reply:
Perhaps we shouldn't be porting these then. Maybe a better solution is to make the functionality accessible as a public method from fuel-core itself. @Voxelot would this seem reasonable to you?